traits/ctraits.c

```diff
@@ -4253,6 +4253,32 @@ _trait_comparison_mode(trait_object *trait, PyObject *args)
     return Py_None;
 }
 
+/*-----------------------------------------------------------------------------
+|  getter for trait comparison mode
++----------------------------------------------------------------------------*/
+
+static PyObject *
+_get_trait_comparison_mode_int(trait_object *trait, PyObject *Py_UNUSED(ignored))
+{
+    PyLongObject *comparison_mode = PyLong_FromLong(2);
+
+    if (trait->flags & TRAIT_NO_VALUE_TEST) {
+        comparison_mode = PyLong_FromLong(0);
+    }
+    else if (trait->flags & TRAIT_OBJECT_ID_TEST) {
+        comparison_mode = PyLong_FromLong(1);
+    }
+
+    if (comparison_mode == NULL) {
+        PyErr_SetString(PyExc_TypeError, "Could not retrieve comparison mode.");
```

Comment:
Just use `return NULL` here - don't set a new exception. If we've got `NULL` at this point, then something else has set an exception with (we hope) a useful exception type and message, so we pass that exception onto the caller by returning `NULL`.